### DISCUSSION TITLE
Regelmäßige Kalibrierladung

### DISCUSSION THREAD

**Discussion by RienSte** [wiggal/GEN24_Ladesteuerung#115] [Ideas]
Hallo @wiggal 

da mein Akku aktuell seit Tagen/Wochen nicht mehr voll wird und mein GEN24 keine Kalibrierladung durchführt (wie gegenteilig im Netz teilweise berichtet), wollte ich fragen ob du so ein Feature implementieren könntest?

Also wenn bspw. in den letzten x Tagen kein SOC von 100% erreicht wurde, dann setze CHARGE_MIN auf y Watt, bis SOC 100% erreicht (für z Minuten, zB 60 Minuten). Danach CHARGE_MIN wieder auf 0 Watt setzen. Ganz cool wäre es, wenn die Akkuschonung noch wahlweise berücksichtigt wird. Dann könnte man guten Gewissens i.d.F. CHARGE_MIN auf den üblichen CHARGE_MAX Wert setzen und trotzdem würde dann bei hohen SOC-Werten die Ladeleistung reduziert.

Gerade bei der Kalibrierladung ist ja sehr wichtig, ab 98-99 % SOC sehr sanft zu laden, damit das BMS ausreichend Zeit hat zu balancen.

Was hältst du davon?
Ich beziehe mich hier auch auf den Thread https://github.com/wiggal/GEN24_Ladesteuerung/discussions/111
Da gehts um etwas ähnliches und ich vermute, der Ursprung ist der selbe.

Danke und SG,
Stephan

**Comment by wiggal** [comment]
Hallo,
aktuell sind leider meine freien Ressourcen voll mit #106 ausgelastet.
Ich würde es aktuell mit der Zwangsladung machen. 
Eine Stunde mit den Watt laden, die bis 90% fehlen, und die nächste Stunde oder zwei mit dementsprechend niedriger Leistung laden, dass der Akku zu 100% kommt.

Vielleicht können wir im Januar nochmal darauf zurückkommen.

Gruß
Wiggal

**Reply by RienSte** [reply]
Guten Morgen,
alles klar, verstehe ich. Das ist ja ein doch ein recht großes Upgrade.

Ich merks mir mal und melde mich einfach nächstes Jahr wieder 😊

**Reply by wiggal** [reply]
Hi,
Ich habe jetzt mal eingebaut, dass auch bei einer Zwangsladung eine Akkuschonung konfiguriert werden kann.
Dazu musst du  `Batterieentlandung_steuern = 2` setzen.
Die Zwangslandung musst du allerdings manuell setzten, ob ich den Automatismus einbaue, weiß ich noch nicht, das icst ziemlich komplex, vermutlich eher nicht.
Wiggal

**Reply by RienSte** [reply]
Hallo @wiggal 

Sorry, habs gar nicht gemerkt dass du hier geantwortet hast. Danke für die Ergänzung, das ist auf jeden Fall schon mal ein Schritt.

Ist es tatsächlich so kompliziert, das einzubauen? Im Grunde müsstest du nur einen Zähler einbauen, der immer irgendwann nach Sonnenuntergang (oder im ersten Durchlauf eines neuen Tages oder so) um eins hoch zählt. Und immer dann, wenn SOC 100% erreicht werden, setzt du den selben Zähler wieder auf 0.
Sollte der Zähler einen definierten Wert erreichen (lt. Konfig), setzt du CHARGE_MIN auf einen Wert, Ladegrund wäre dann "Kalibrierladung". Sind die 100% SOC erreicht wird der Zähler (durch die o.g. generelle Abfrage) wieder auf 0 gesetzt und somit fällt auch das Kriterium für die Kalibierladung wieder weg und es wird wieder CHARGE_MIN aus den anderen Code-Teilen (Zwangsladung, DynPriceing, etc) geschrieben.

Je nachdem wie dein Code aufgebaut ist, kann der Aufwand natürlich schon recht groß sein, aber rein die Logik erscheint mir nicht allzu kompliziert.

Schöne Grüße,
Stephan

**Reply by wiggal** [reply]
Ja, die Logik ist nicht so schwierig, da gebe ich dir recht. 
Aber das Zusammenspiel mir den anderen Zwangsladeskripten und Bedingungen macht das ganze schon sehr kompliziert.
Und den Zähler müsste man auch schreiben ....

Also aktuell werde ich in der Richtung nichts machen, sehe auch eher keinen Bedarf, bei mir macht der Akku schon immer wieder auch das Balancing, aktuell auch bei 80%.

![grafik](https://github.com/user-attachments/assets/0107a32c-6ddb-4b8d-88d8-bc9e51aaa1b2)

**Reply by RienSte** [reply]
Alles klar, danke auch für den Screenshot. Mit welchem Tool liest du die Zellspannungen aus?

**Reply by wiggal** [reply]
https://www.photovoltaikforum.com/thread/237830-byd-logger-ollis-tool-download/

**Reply by RienSte** [reply]
Danke dir!!
